### PR TITLE
Owlot/dw 734/svc storage user group

### PR DESCRIPTION
**Pull request by owlot**
Adding new `user`, `group` and `user_group` tables.
Also adds grpc services for linking groups to users and users to groups.

### CODE REVIEW

File: client-grpc/src/lib.rs
Comment (ServiceDog) [inline client-grpc/src/lib.rs:1]: off-topic, but could you make the 'geo_types' `pub` so I can use the Polygon::Centroid function in svc-cargo?

I'm just returning a list of lat,lons to the customer for each vertiport, don't need to deliver the Polygon to them

File: server/src/resources/user/mod.rs [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        assert!(result.is_ok());
+        if let Ok((sql_fields, validation_result)) = result {
+            println!("{:?}", sql_fields);
+            println!("{:?}", validation_result);
+            assert_eq!(validation_result.success, true);
+        }
+    }
+
+    #[test]
+    fn test_scanner_invalid_data() {
```
Comment (ServiceDog) [inline server/src/resources/user/mod.rs:124]: some mentions to `scanner` in here, I think a copy pasta

File: proto/svc-storage-grpc-user.proto [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    google.protobuf.FieldMask mask = 3;
+}
+
+// Data struct with user data
+// PI data will be stored in a separate database/ service
+message Data {
+    // the User's authentication method
+    AuthMethod auth_method = 2;
+    // the User's preferred display name
+    string display_name = 3;
```
Comment (ServiceDog) [inline proto/svc-storage-grpc-user.proto:57]: This is a good start for now.

Should Hangar and Cargo Shipper UI user settings be in a different table, or are those settings in local cookies?

We can add preferred contact method in another PR

File: proto/svc-storage-grpc-group.proto [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+// Struct containing a `list` of group [Vec\<Object\>]
+message List {
+    // array/vector of group items
+    repeated Object list = 1;
+}
+
+// Struct used to link users to group
+message GroupUsers {
+    // `id` [String] in [Uuid](uuid::Uuid) format
+    string id = 1;
```
Comment (ServiceDog) [inline proto/svc-storage-grpc-group.proto:58]: would this be better as group_id and user_ids, or are you keeping to a uniform 'id' and 'other_id_list' across all link table messages

**Review by ServiceDog — CHANGES_REQUESTED** [review]

File: proto/svc-storage-grpc-user.proto (hunk 2 above)
Comment (owlot) [reply proto/svc-storage-grpc-user.proto:57]: Settings will be going into a special 'fields' table which will make it easier to add/ remove settings later on.
I've used this method in the past, and it worked pretty well, it'll be something like this:
Selection_240

We then add a module into svc-storage where resources can be marked with a 'has_fields' setting so you would get the possibility to get/update/set these fields through grpc as well.

File: proto/svc-storage-grpc-group.proto (hunk 3 above)
Comment (owlot) [reply proto/svc-storage-grpc-group.proto:58]: I'm keeping it uniform yes, the fields are the same as we have for itinerary_flight_plan.

File: server/src/resources/user/mod.rs (hunk 1 above)
Comment (owlot) [reply server/src/resources/user/mod.rs]: fixed!

File: client-grpc/src/lib.rs
Comment (owlot) [reply client-grpc/src/lib.rs:1]: Fixed in 496d83c16f8a47bf941318f1073be6681375d2d7

**Pull request merged**